WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/usecases/FetchScanStateUseCase.kt

```diff
@@ -0,0 +1,62 @@
+package org.wordpress.android.ui.jetpack.scan.usecases
+
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.scan.ScanStateModel
+import org.wordpress.android.fluxc.store.ScanStore
+import org.wordpress.android.fluxc.store.ScanStore.FetchScanStatePayload
+import org.wordpress.android.ui.jetpack.scan.usecases.FetchScanStateUseCase.FetchScanState.Failure
+import org.wordpress.android.ui.jetpack.scan.usecases.FetchScanStateUseCase.FetchScanState.Success
+import org.wordpress.android.util.NetworkUtilsWrapper
+import javax.inject.Inject
+
+const val FETCH_SCAN_STATE_DELAY_MILLIS = 5000L
+
+class FetchScanStateUseCase @Inject constructor(
+    private val networkUtilsWrapper: NetworkUtilsWrapper,
+    private val scanStore: ScanStore
+) {
+    suspend fun fetchScanState(
+        site: SiteModel,
+        polling: Boolean = true,
+        delayInMs: Long = FETCH_SCAN_STATE_DELAY_MILLIS
+    ): Flow<FetchScanState> = flow {
+        if (!networkUtilsWrapper.isNetworkAvailable()) {
+            emit(Failure.NetworkUnavailable)
+            return@flow
+        }
+
+        val result = scanStore.fetchScanState(FetchScanStatePayload(site))
+        if (result.isError) {
+            emit(Failure.RemoteRequestFailure)
+            return@flow
+        } else {
+            val scanStateModel = scanStore.getScanStateForSite(site)
+            if (scanStateModel != null) {
+                emit(Success(scanStateModel))
+                if (polling && scanStateModel.state == ScanStateModel.State.SCANNING) {
+                    fetchScanState(site = site, polling = true).collect {
+                        emit(it)
+                        delay(delayInMs)
+                    }
```

Comment:
This is a tricky part with a recursive call to the `fetchScanState` function during polling. We might want to replace it with a `while` loop.